clients/drcachesim/tools/invariant_checker.h

```diff
@@ -126,19 +126,33 @@ class invariant_checker_t : public analysis_tool_t {
         memtrace_stream_t *stream = nullptr;
         memref_t prev_entry_ = {};
         memref_t prev_prev_entry_ = {};
-        memref_t prev_instr_ = {};
-        std::unique_ptr<instr_autoclean_t> prev_instr_decoded_ = nullptr;
         memref_t prev_xfer_marker_ = {}; // Cleared on seeing an instr.
         memref_t last_xfer_marker_ = {}; // Not cleared: just the prior xfer marker.
         uintptr_t prev_func_id_ = 0;
         // We keep track of return addresses of nested function calls.
         std::stack<addr_t> retaddr_stack_;
         uintptr_t trace_version_ = 0;
+        // Struct to store decoding related attributes.
+        struct decoding_info_t {
+            bool has_valid_decoding = false;
+            bool is_syscall = false;
+            bool writes_memory = false;
+            bool is_predicated = false;
+            uint num_memory_read_access = 0;
+            uint num_memory_write_access = 0;
+            addr_t branch_target;
+        };
+        struct instr_info_t {
+            memref_t memref;
```

Comment:
I would zero this one out `= {}` and then the default constructor is all we need for this struct.